recurring_contract/models/recurring_contract.py

```diff
@@ -182,39 +185,54 @@ def unlink(self):
     #                             PUBLIC METHODS                             #
     ##########################################################################
     @api.multi
-    def clean_invoices(self, since_date=None, to_date=None, keep_lines=None):
+    def clean_invoices(self, since_date=None, to_date=None, keep_lines=None, **kwargs):
         """ By default, launch asynchronous job to perform the task.
             Context value async_mode set to False can force to perform
             the task immediately.
         """
         if self.env.context.get('async_mode', True):
-            self.with_delay()._clean_invoices(since_date, to_date, keep_lines)
+            self.with_delay()._clean_invoices(since_date, to_date, keep_lines, **kwargs)
         else:
-            self._clean_invoices(since_date, to_date, keep_lines)
+            self._clean_invoices(since_date, to_date, keep_lines, **kwargs)
 
     def rewind_next_invoice_date(self):
         """ Rewinds the next invoice date of contract after the last
         generated invoice. No open invoices exist after that date. """
         gen_states = self.env['recurring.contract.group']._get_gen_states()
+
+        res = self.env["account.invoice"]
+
         for contract in self:
             if contract.state in gen_states:
-                last_invoice_date = max([
-                    line.invoice_id.date_invoice for
-                    line in contract.invoice_line_ids
-                    if line.state in ('open', 'paid')] or [False])
+                last_invoice_date, is_paid = min(
```

Comment:
Why are you taking the next future invoice instead of looking for the last invoice like it was previously done?
If that change is required, then I would suggest to rename the variable `last_invoice_date` to something more explanatory.

Reply:
We decided, I remember, to regenerate invoice with date >= to today. I first though about regenerating all opened invoices but you explained to me that this could lead to some issue. 

With this new regeneration approach multiple open invoices could be changed if they are still open.

I do agree that the name of the variable could be different.

Reply:
Yes I now remember this discussion. If you find a better name I would be happy. Not mandatory change though. Maybe you could also rework the comment of the method for better explaining what it's doing.